### ISSUE TITLE
New experiment component

### ISSUE THREAD

**Issue by ddelpiano** [MetaCell/NetPyNE-UI#231]
![Screenshot 2021-03-26 at 10.18.40.png](https://images.zenhubusercontent.com/5b0293e84b5806bc2bd3f170/43cf1f93-df11-4380-8735-c841c168dbd8)

The focus of this card is the implementation of the new experiment component within the flexlayout implementation, few notes:
- this component is part of the experiment manager, is a transition triggered by the link/button at the bottom of the experiment manager
- this implementation does not cover the tree autocomplete component and the paramenter picker component, there will be dedicated cards for them, at this stage the parameter will be defined/selected through the entire string in what will be a normal input text, for the autocomplete tree component there is a follow up card to implement just that.

2 notes: 
- currently the mockups are missing the group parameter functionality, so please refer to the mockups before to start the implementation.
- If needed this card can overlap with the next sprint since the estimate is above the current allocation.

@ddelpiano  the mockups are missing the initConfig functionality as well, it will be a simplified version of the exploration parameters: select a simConfig variable and set **one** value, no range or list type, just overwriting configuration.

@vidhya-metacell feel free to ask if we missed anything in the description of the card that you are expecting to have to implement this.

Thanks.

**Comment by lrebscher** [comment]
You can find below an example payload for creating and editing a new Experiment.
It's basically a reduced version of the Experiments that are returned as dummy data.

I created a component `EditExperiment.js` for add/edit of Experiment that loads an experiment from the Redux store.

```javascript
{
    "name": "EI Populations",
    "params": [
        {
            "label": "weight",
            "mapsTo": "netParams.connParams['E->E']['weight']",
            "type": "list",
            "values": [
                1,
                2,
                3,
                4
            ],
            "min": null,
            "max": null,
            "step": null
        },
        {
            "label": "probability",
            "mapsTo": "",
            "type": "range",
            "values": null,
            "min": 0.3,
            "max": 1,
            "step": 0.2
        }
    ],
    "initConfig": {},
}
```

This will be the same payload that you will use to update the experiments, the views behave at the moment in the same way.

You won't be calling any backend methods directly, this will happen in the Redux middleware or in a dedicated hook, for now just use dummy functions to submit and edit data.

**Comment by ddelpiano** [comment]
@vidhya-metacell please be aware that this component has been subject of discussion for what it concern the group parameter feature and @paulfaivret is doing some more changes in the mockups.

**Comment by ddelpiano** [comment]
@paulfaivret pinging you here to keep vidhya and lucas in the loop as well, just doing a copy paste from our chat.

- On the create new experiment, we noticed a discrepancy in the footer of the component when the group parameter is there and when it is not, could you advice on that please?
- same component, we can see that the group parameter card/dark square once is up and once is down within the component, could explain us the flow please?
- Error handling for these new components, how should we handle that? Dialog or we show the error message in the input text? Could you give us few example in the mockups please?

**Comment by paulfaivret** [comment]
@ddelpiano This component was not completely done, so I've updated it and here are the answers :
- The idea was to make the background of the button group appear only when it is needed (i.e. when there is scrollable content that goes under the group) - but that ultimately complicates the implementation, so I updated the Figma file and the background now appears all the time.
- Just a bug on this one, it's fixed
- If the error is related to an input field, it goes into 'error' state (I created one on Figma just in case). If the error is related to the backend, it is not displayed in the component directly but as a classic backend error. If the user adds only one parameter to a parameter group, the interface does not present an error but considers the parameter alone without the group functionalities and a message is displayed (see on the Figma file)
